modules/platform/nuxeo-platform-csv-export/src/main/java/org/nuxeo/ecm/platform/csv/export/io/DocumentPropertyCSVWriter.java

```diff
@@ -167,19 +177,26 @@ protected String getTranslatedValue(Session session, String schema, String lang,
             label = (String) entry.getProperty(schema, "label");
         } catch (PropertyNotFoundException e) {
             try {
-                // Check if it comes from a l10nxvocabulary, and return this value if it exists,
-                // or else return the id
-                return (String) entry.getProperty(schema, "label_en");
+                // Check if it comes from a l10nxvocabulary, and return this value if it exists, or else return the id
+                return replaceCarriageReturn((String) entry.getProperty(schema, "label_en"));
             } catch (PropertyNotFoundException e1) {
-                return value;
+                return replaceCarriageReturn(value);
             }
         }
         Locale locale = lang != null ? Locale.forLanguageTag(lang) : ctx.getLocale();
-        return I18NUtils.getMessageString("messages", label, new Object[0], locale);
+        return replaceCarriageReturn(I18NUtils.getMessageString("messages", label, new Object[0], locale));
     }
 
     protected void writeUnsupported(Type type, CSVPrinter printer) throws IOException {
         printer.print(String.format("type %s is not supported", type.getName()));
     }
 
+    protected String replaceCarriageReturn(String value) {
+        String replacement = ctx.getParameter(CARRIAGE_RETURN_REPLACEMENT_CTX_DATA);
+        if (value == null || replacement == null) {
+            return value;
+        }
+        return NEWLINE.matcher(value).replaceAll(replacement);
```

Comment:
In order to avoid escaping in the caller and avoid too much work (not quote when there's nothing to replace, which is common) we could do:

    Matcher m = NEWLINE.matcher(value);
    if (m.find()) {
        value = m.replaceAll(Matcher.quoteReplacement(replacement));
    }
    return value;